### PR TITLE
Data mapper integration updates for #602.

### PR DESCRIPTION
**Pull request by jbakermd**
Updated integration between Data Mapper and Syndesis UI per #602

### CODE REVIEW

**Comment by gashcrumb** [comment]
@jbakermd looks like this PR still needs to update the `package.json` to point to the new repo and run `yarn install`

**Comment by gashcrumb** [comment]
Also really digging the simplifications for initializing the component, thanks a bunch for that!

**Comment by jbakermd** [comment]
Good catch, forgot to copy that change over from my scratch directory where the work was done.

**Comment by jbakermd** [comment]
FYI I see a red X by the https://github.com/syndesisio/syndesis-ui/pull/608/commits/719f87bc32c71f358877e1b4f67590c5a6a93031 commit here saying a build failed. Hopefully that package.json change will rectify the issue.

**Comment by jbakermd** [comment]
Wait one second, I see a typo.

File: package.json [hunk 1]
```diff
@@ -66,7 +66,7 @@
     "patternfly": "3.20.0",
     "pluralize": "5.0.0",
     "rxjs": "5.4.1",
-    "syndesis.data.mapper": "https://github.com/syndesisio/syndesis-datamapper-ui.git#9476d91ec03c0e4885f3005876c67216130ef98d",
+    "syndesis.data.mapper": "https://github.com/atlasmap/atlasmap-ui.git",
```
Comment (gashcrumb) [inline package.json]: I think you'll still need the commit hash on this URL too, but let's see...

Comment (jbakermd) [inline package.json]: ok i can do that, let me find the latest hash there and make sure it builds local.

Comment (jbakermd) [inline package.json]: wait a moment, after setting that i'm receiving build errors:

`
yarn start
yarn start v0.22.0
$ npm run build

> syndesis.data.mapper@1.0.0 build /Users/jason/Documents/code/md/ipaasui/syndesis-ui/node_modules/syndesis.data.mapper
> tsc -p ./

../@types/lodash/index.d.ts(19449,15): error TS2428: All declarations of 'WeakMap' must have identical type parameters.
../typescript/lib/lib.es2015.collection.d.ts(45,11): error TS2428: All declarations of 'WeakMap' must have identical type parameters.
../typescript/lib/lib.es2015.iterable.d.ts(124,11): error TS2428: All declarations of 'WeakMap' must have identical type parameters.
../typescript/lib/lib.es2015.symbol.wellknown.d.ts(133,11): error TS2428: All declarations of 'WeakMap' must have identical type parameters.
`

Comment (jbakermd) [inline package.json]: Ok a clean build fixed that problem, committing the update with added hash.

**Comment by jbakermd** [comment]
One more quick review please @gashcrumb - I updated the package.json dependency to point to a specific hash of datamapper.

**Comment by gashcrumb** [comment]
@jbakermd have some lint errors to look at too:

```
src/app/integrations/edit-page/step-configure/data-mapper-host.component.ts[35, 27]: trailing whitespace
src/app/integrations/edit-page/step-configure/data-mapper-host.component.ts[90, 52]: trailing whitespace
src/app/integrations/edit-page/step-configure/data-mapper-host.component.ts[92, 51]: trailing whitespace
src/app/integrations/edit-page/step-configure/data-mapper-host.component.ts[94, 52]: trailing whitespace
src/app/integrations/edit-page/step-configure/data-mapper-host.component.ts[96, 45]: trailing whitespace
src/app/integrations/edit-page/step-configure/data-mapper-host.component.ts[89, 5]: Forbidden 'var' keyword, use 'let' or 'const' instead
src/app/integrations/edit-page/step-configure/data-mapper-host.component.ts[111, 5]: Forbidden 'var' keyword, use 'let' or 'const' instead
src/app/integrations/edit-page/step-configure/data-mapper-host.component.ts[89, 19]: " should be '
src/app/integrations/edit-page/step-configure/data-mapper-host.component.ts[91, 28]: " should be '
src/app/integrations/edit-page/step-configure/data-mapper-host.component.ts[91, 70]: " should be '
src/app/integrations/edit-page/step-configure/data-mapper-host.component.ts[93, 28]: " should be '
src/app/integrations/edit-page/step-configure/data-mapper-host.component.ts[93, 69]: " should be '
src/app/integrations/edit-page/step-configure/data-mapper-host.component.ts[95, 28]: " should be '
src/app/integrations/edit-page/step-configure/data-mapper-host.component.ts[95, 70]: " should be '
src/app/integrations/edit-page/step-configure/data-mapper-host.component.ts[97, 28]: " should be '
src/app/integrations/edit-page/step-configure/data-mapper-host.component.ts[149, 11]: Missing trailing comma
src/app/integrations/edit-page/step-configure/data-mapper-host.component.ts[161, 18]: Missing trailing comma
```

**Comment by jbakermd** [comment]
Cool. Fixed those.

**Comment by gashcrumb** [comment]
@jbakermd awesome, thanks!

**Pull request merged**